src/sidebar/store/modules/sidebar-panels.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+const util = require('../util');
+
+function init() {
+  return {
+    /*
+     * The `panelName` of the currently-active sidebar panel.
+     * Only one `panelName` may be active at a time, but it is valid (though not
+     * the standard use case) for multiple `SidebarPanel` components to share
+     * the same `panelName`—`panelName` is not intended as a unique ID/key.
+     *
+     * e.g. If `activeSidebarPanel` were `foobar`, all `SidebarPanel` components
+     * with `panelName` of `foobar` would be active, and thus visible.
+     *
+     */
+    activeSidebarPanel: null,
+  };
+}
+
+const update = {
+  OPEN_SIDEBAR_PANEL: function(state, action) {
+    return { activeSidebarPanel: action.panelName };
+  },
+
+  CLOSE_SIDEBAR_PANEL: function(state, action) {
+    let activeSidebarPanel = state.activeSidebarPanel;
```

Comment:
what use case do we have where we don't pass the right name in?

In other words, would it be bad if we just always set the active panel to `null`?

Reply:
I alluded to this laxness in the description for this PR (and the code overall needs more commenting). Basically, the `store` doesn't care a toot if a panel or panels with this name exists. It's dumb. It merely keeps track of what it's told (a simple string). Whether or not there's an actual corresponding panel, it cares not.

There may be a better way to do this, and I hemmed and hawed a while over trying to have some sort of "registry" of actual panels, but this felt both more complex (i.e. bug-prone) and could lead to painful maintenance.

What I did for now to slightly alleviate this mysteriousness is use the `ui-constants` module to define the one known panel name, so that components could make use of that constant and not be flinging random strings around.